from_cpython/Lib/test/test_dict.py

```diff
@@ -556,7 +555,7 @@ def __eq__(self, other):
         d[X()] = 5
         # now trigger a resize
         resizing = True
-        d[9] = 6
+        # d[9] = 6
```

Comment:
This test cause pyston hang on. Seems the `BoxAndHash::Eq` has problem, then cause DenseMap resize hang on at last. But I can't go further, @undingen , would you mind to give some helps?